src/vs/workbench/contrib/terminal/node/terminalProfiles.ts

```diff
@@ -0,0 +1,154 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as fs from 'fs';
+import * as platform from 'vs/base/common/platform';
+import { coalesce } from 'vs/base/common/arrays';
+import { normalize, basename } from 'vs/base/common/path';
+import { enumeratePowerShellInstallations } from 'vs/base/node/powershell';
+import { getWindowsBuildNumber } from 'vs/platform/terminal/node/terminalEnvironment';
+import { ITerminalProfile } from 'vs/workbench/contrib/terminal/common/terminal';
+import * as cp from 'child_process';
+
+export interface IStatProvider {
+	stat(path: string): boolean,
+	lstat(path: string): boolean
+}
+
+export function detectAvailableShells(detectWslShells?: boolean, statProvider?: IStatProvider): Promise<ITerminalProfile[]> {
```

Comment:
We should rename the `Select Default Shell` command to `Select Default Profile`, we'll keep the command name the same though for backwards compat.